frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -1118,7 +1100,19 @@ mod tests {
 						distribution: vec![(10, PerU16::one())],
 					}],
 				};
-				let (solution, witness) = MultiPhase::prepare_election_result(result).unwrap();
+
+				let RoundSnapshot { voters, targets } = MultiPhase::snapshot().unwrap();
+				let desired_targets = MultiPhase::desired_targets().unwrap();
+
+				let (raw, score, witness) =
+					Miner::<Runtime>::prepare_election_result_with_snapshot(
+						result,
+						voters.clone(),
+						targets.clone(),
```

Comment:
Are those `clone`s necessary? Can't we change the `prepare_election_result_with_snapshot` to take `&[T]` instead of a `Vec<T>` for those? It doesn't seem to actually need those owned from what I can see?

Reply:
in the production code `prepare_election_result_with_snapshot` is the last place that we use these values so IMO it is cleaner to consume everything. Clones are only needed in tests.

Reply:
Ah okay, that's fine then.